pkg/cloudclient/aws/private.go

```diff
@@ -446,17 +490,38 @@ func (c *Client) verifyDns(ctx context.Context, vpcID string) *output.Output {
 		Attribute: "enableDnsSupport",
 		VpcId:     aws.String(vpcID),
 	})
+	if dnsSprtErr != nil {
+		var aerr smithy.APIError
+		if errors.As(dnsSprtErr, &aerr) {
+			switch {
+			case aerr.ErrorCode() == "UnauthorizedOperation":
+				c.output.AddError(errors.New("missing required permission ec2:DescribeVpcAttribute"))
+				return &c.output
+			}
+		}
+
+		c.output.AddError(fmt.Errorf("error calling ec2:DescribeVpcAttribute: %w", dnsSprtErr))
```

Comment:
Sorry for being late to the party, but just a small question:
currently, AddError() adds the error as generic: https://github.com/openshift/osd-network-verifier/blob/main/pkg/output/output.go#L20 
would it make sense to refactor here to a more elastic form that could accept generic/egressURL/aws permission-related errors?

Reply:
Yeah, that sounds like a good idea. It would make the code a little DRY-er, I'll see if I can try that in my next PR

Reply:
Followed up with https://github.com/openshift/osd-network-verifier/pull/130